cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -111,26 +111,30 @@ LedgerLoop:
 			}
 		}
 
-		// Initialise tx reader and move it to start idx.
+		// Initialise tx reader.
 		reader, err := ingest.NewLedgerTransactionReaderFromLedgerCloseMeta(h.networkPassphrase, ledger)
 		if err != nil {
 			return GetTransactionsResponse{}, &jrpc2.Error{
 				Code:    jrpc2.InvalidParams,
 				Message: err.Error(),
 			}
 		}
-		err = reader.Seek(int(start.TxIdx))
+
+		// Move the reader to specific tx idx
+		startTxIdx := 0
+		if ledgerSeq == start.LedgerSequence {
+			startTxIdx = int(start.TxIdx)
+		}
+		err = reader.Seek(startTxIdx)
 		if err != nil {
-			return GetTransactionsResponse{}, &jrpc2.Error{
-				Code:    jrpc2.InvalidParams,
-				Message: err.Error(),
-			}
+			// Seek returns EOF so we can move onto next ledger
+			continue
 		}
 
 		// Decode transaction info from ledger meta
 		txCount := ledger.CountTransactions()
-		for i := start.TxIdx; i < uint32(txCount); i++ {
-			cursor = transactions.NewCursor(ledger.LedgerSequence(), i, 0)
+		for i := startTxIdx; i < txCount; i++ {
+			cursor = transactions.NewCursor(ledger.LedgerSequence(), uint32(i), 0)
```

Comment:
`< txCount` is wrong, right? It should be `txCount - startTxIdx` (maybe +/-1? check me). Or, better might just be to iterate from `startTxIdx` until you get `io.EOF` from the reader

Reply:
@Shaptic No the `txCount` would work here because only the startIdx is shifting towards the end. And we get the startIdx from the cursor which is the last txIdx we read from it. The indexing is not changing here.

Reply:
suppose startTxIdx = 5 and txCount = 10, what I'm saying is that this will definitely encounter EOF after just 5 iterations and return an error (EOF check should be a `break` I think)

Reply:
nvm I can't read